integration_tests/src/fixture_generator/generator.rs

```diff
@@ -0,0 +1,181 @@
+
+type Blake3 = Blake3_256<BaseElement>;
+
+/// Directory state comprises all database records at a particular epoch.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct State {
+    pub epoch: u32,
+    pub records: Vec<DbRecord>,
+}
+
+/// Delta comprises all updates published to the directory to advance to an
+/// epoch.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Delta {
+    pub epoch: u32,
+    pub updates: Vec<(AkdLabel, AkdValue)>,
+}
+
+/// Metadata about the output, including arguments passed to this tool and
+/// the tool version.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Metadata {
+    pub args: Args,
+    pub version: String,
+}
+
+// "@" has to be separated from "generated" or linters might ignore this file
+const HEADER_COMMENT: &str = concat!(
+    "@",
+    "generated This file was automatically generated by \n\
+    the fixture generator tool with the following command:\n\n\
+    cargo run -- \\"
+);
+const METADATA_COMMENT: &str = "Metadata";
+const STATE_COMMENT: &str = "State - Epoch";
+const DELTA_COMMENT: &str = "Delta - Epoch";
+
+pub async fn run() {
+    let args = Args::parse();
+    let mut rng = OsRng;
+
+    // args assertions
+    assert!(args.max_updates >= args.min_updates);
+    assert!(args
+        .capture_states
+        .as_ref()
+        .map_or(true, |states| states.iter().max().unwrap() <= &args.epochs));
+    assert!(args
+        .capture_deltas
+        .as_ref()
+        .map_or(true, |deltas| deltas.iter().max().unwrap() <= &args.epochs));
+
+    // process users
+    let mut user_map = HashMap::new();
+    for user in &args.users {
+        let mut events_map = HashMap::new();
+        for event in &user.events {
+            events_map.insert(event.epoch, event.value.clone());
+        }
+        user_map.insert(user.label.clone(), events_map);
+    }
+
+    // initialize writer
+    let buffer: Box<dyn Write> = if let Some(ref file_name) = args.out {
+        Box::new(File::create(file_name).unwrap())
+    } else {
+        Box::new(std::io::stdout())
+    };
+    let mut writer = YamlWriter::new(buffer);
+
+    // write raw args as comment
+    let raw_args = format!(
+        " {}",
+        env::args().skip(1).collect::<Vec<String>>().join(" ")
+    );
+    writer.write_comment(HEADER_COMMENT);
+    raw_args
+        .split(" -")
+        .skip(1)
+        .map(|arg| format!("  -{} \\", arg))
+        .for_each(|comment| writer.write_comment(&comment));
+
+    // write fixture metadata
+    let comment = METADATA_COMMENT.to_string();
+    let metadata = Metadata {
+        args: args.clone(),
+        version: env!("CARGO_PKG_VERSION").to_string(),
+    };
+    writer.write_line();
+    writer.write_comment(&comment);
+    writer.write_object(metadata);
+
+    // initialize directory
+    let db = akd::storage::memory::AsyncInMemoryDatabase::new();
+    let vrf = akd::ecvrf::HardCodedAkdVRF {};
```

Comment:
So potentially providing their own VRF is fine, but for this tool why would we want to utilize another database impl? Just to validate other implementations? To "generate" the change file I would think in-memory is the easiest, unless we're going to look at GB of memory (i.e. generating a huge file), in which case we'll have problems serializing and deserializing the Struct anyways. 

Personally I don't think we need other db impls. Now reading **in** data is a different topic, in that case it should be supported on all db's in order to support bootstrapping I think.

Reply:
> unless we're going to look at GB of memory (i.e. generating a huge file)

This was exactly what I had in mind, using a DB to generate a larger fixture file. But you're absolutely right that the reader would have problems deserializing which I failed to consider :P Let's go with not supporting other DBs for the generation.